authenticator.py

```diff
@@ -74,10 +80,18 @@ def authenticate(cls, request):
             get_nio_logger("Basic.Authenticator").error(msg)
             raise Unauthorized(msg)
 
-        # check password
-        if base64_encode(password) != user.get('password'):
-            msg = "Password is invalid."
-            get_nio_logger("Basic.Authenticator").error(msg)
-            raise Unauthorized(msg)
-
-        return User(name=username)
+        stored_pwd = user.get('password')
+        # validate bcrypt hashed password
+        if stored_pwd.startswith('$2b$'):
```

Comment:
Any chance the `$2b$` format could change in the future? I know `$2a$` was the format for a previous version of bcrypt. Wondering if there is a better way to check if it is hashed. Maybe even `startswith('$2') ?

Reply:
It has gone through some patches in the past from like `$2a$` and then a `$2y$` at one point. What about some sort of regex match: `^\$2[ayb]\$$`?

Reply:
And this could be updated as security patches for bcrypt are released as well